--- plugin/x/protocol/plugin/message_field_chain.cc.orig	2025-10-15 19:18:16.802323135 +0100
+++ plugin/x/protocol/plugin/message_field_chain.cc	2025-10-15 19:19:58.206055544 +0100
@@ -58,7 +58,7 @@
 
   // Check against cycles in Message dependencies graph
   const bool was_node_visited =
-      message && 0 != m_types_done.count(message->full_name());
+      message && 0 != m_types_done.count(std::string(message->full_name()));
 
   if (nullptr == message || was_node_visited || 0 == message->field_count()) {
     m_output_file->append_chain(m_context, chain);
@@ -74,7 +74,7 @@
 
 void Message_field_chain::end_validate_field(const FieldDescriptor *field,
                                              const Descriptor *message) {
-  m_types_done.erase(message->full_name());
+	m_types_done.erase(std::string(message->full_name()));
   const auto position = m_chain.find_last_of(".");
 
   if (std::string::npos != position) m_chain.resize(position);
