diff --git a/plugin/x/protocol/plugin/encoder_file_output.h b/plugin/x/protocol/plugin/encoder_file_output.h
index 0c42382e..ef9da93d 100644
--- a/plugin/x/protocol/plugin/encoder_file_output.h
+++ b/plugin/x/protocol/plugin/encoder_file_output.h
@@ -97,7 +97,7 @@ class Encoder_file_output : public File_output {
       bool is_reserved = false;
       const auto field = message->field(i);
       const auto field_tag = std::to_string(field->number());
-      const auto field_name = get_cpp_field_name(field->name(), &is_reserved);
+      const auto field_name = get_cpp_field_name(std::string(field->name()), &is_reserved);
 
       if (is_reserved) {
         write_to_context(context,
@@ -129,7 +129,7 @@ class Encoder_file_output : public File_output {
 
       // Check if we there is a message containing Server message ID
       if (0 == m_used_message_ids.count(enum_value->number())) {
-        std::string value = "  " + enum_value->name() + " = " +
+        std::string value = "  " + std::string(enum_value->name()) + " = " +
                             std::to_string(enum_value->number());
 
         values.push_back(value);
